File: backend/src/models/Product.js
const mongoose = require('mongoose');

const productSchema = new mongoose.Schema({
  name: { type: String, required: true, trim: true },
  categoryId: { type: mongoose.Schema.Types.ObjectId, ref: 'Category', required: true },
  price: { type: Number, required: true, min: 0 },
  gstRate: { type: Number, default: 5 }, // GST %
  image: { type: String, default: '' },
  description: { type: String, default: '' },
  isAvailable: { type: Boolean, default: true },
  isVeg: { type: Boolean, default: true },
  sku: { type: String, default: '' }
}, { timestamps: true });

productSchema.index({ categoryId: 1, name: 1 });

module.exports = mongoose.model('Product', productSchema);

File: frontend_flutter/lib/models/product.dart
class ProductModel {
  final String id;
  final String name;
  final double price;
  final double gstRate;
  final bool isAvailable;
  final bool isVeg;

  ProductModel({
    required this.id,
    required this.name,
    required this.price,
    required this.gstRate,
    required this.isAvailable,
    required this.isVeg,
  });

  factory ProductModel.fromJson(Map<String, dynamic> json) {
    return ProductModel(
      id: json['_id'] ?? '',
      name: json['name'] ?? '',
      price: (json['price'] ?? 0).toDouble(),
      gstRate: (json['gstRate'] ?? 5).toDouble(),
      isAvailable: json['isAvailable'] ?? true,
      isVeg: json['isVeg'] ?? true,
    );
  }
}
